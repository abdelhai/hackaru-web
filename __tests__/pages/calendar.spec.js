import MockDate from 'mockdate';
import { Store } from 'vuex-mock-store';
import Factory from '@/__tests__/__setups__/factory';
import Calendar from '@/pages/calendar';
import { parse } from 'date-fns';

describe('Calendar', () => {
  let wrapper;
  let factory;

  MockDate.set('2019-01-31T01:23:45');

  const $store = new Store({});

  beforeEach(() => {
    factory = new Factory(Calendar, {
      stubs: {
        'infinite-slider': {
          template: '<div><slot /></div>',
          methods: {
            slideLeft: jest.fn(),
            slideRight: jest.fn()
          }
        }
      },
      mocks: {
        $store
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  it('dispatch activities/getActivities', () => {
    factory.shallow();
    expect($store.dispatch).toHaveBeenCalledWith('activities/getActivities', {
      start: parse('2019-01-27T00:00:00'),
      end: parse('2019-02-02T23:59:59.999')
    });
  });

  describe('when period is daily', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setData({ index: 0 });
    });

    it('has days correctly', () => {
      expect(wrapper.vm.days).toEqual([parse('2019-01-31T00:00:00')]);
    });
  });

  describe('when period is weekly', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setData({ index: 1 });
    });

    it('has days correctly', () => {
      expect(wrapper.vm.days).toEqual([
        parse('2019-01-27T00:00:00'),
        parse('2019-01-28T00:00:00'),
        parse('2019-01-29T00:00:00'),
        parse('2019-01-30T00:00:00'),
        parse('2019-01-31T00:00:00'),
        parse('2019-02-01T00:00:00'),
        parse('2019-02-02T00:00:00')
      ]);
    });
  });

  describe('when call togglePeriod', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setData({ index: 1 });
      wrapper.vm.togglePeriod(parse('2019-01-27T00:00:00'));
    });

    it('toggle period', () => {
      expect(wrapper.vm.index).toBe(0);
    });

    it('set date', () => {
      expect(wrapper.vm.date).toEqual(parse('2019-01-27T00:00:00'));
    });
  });

  describe('when slide left and period is daily', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setData({ index: 0 });
      wrapper.find({ ref: 'slider' }).vm.$emit('slide-left');
    });

    it('set previous date', () => {
      expect(wrapper.vm.date).toEqual(parse('2019-01-30T00:00:00'));
    });
  });

  describe('when slide left and period is weekly', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setData({ index: 1 });
      wrapper.find({ ref: 'slider' }).vm.$emit('slide-left');
    });

    it('set previous week', () => {
      expect(wrapper.vm.date).toEqual(parse('2019-01-20T00:00:00'));
    });
  });

  describe('when slide right and period is daily', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setData({ index: 0 });
      wrapper.find({ ref: 'slider' }).vm.$emit('slide-right');
    });

    it('set next date', () => {
      expect(wrapper.vm.date).toEqual(parse('2019-02-01T00:00:00'));
    });
  });

  describe('when slide right and period is weekly', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setData({ index: 1 });
      wrapper.find({ ref: 'slider' }).vm.$emit('slide-right');
    });

    it('set next week', () => {
      expect(wrapper.vm.date).toEqual(parse('2019-02-03T00:00:00'));
    });
  });

  describe('when click left button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find({ ref: 'header' }).vm.$emit('left');
    });

    it('call slideLeft', () => {
      expect(
        factory.options.stubs['infinite-slider'].methods.slideLeft
      ).toHaveBeenCalled();
    });
  });

  describe('when click right button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find({ ref: 'header' }).vm.$emit('right');
    });

    it('call slideRight', () => {
      expect(
        factory.options.stubs['infinite-slider'].methods.slideRight
      ).toHaveBeenCalled();
    });
  });
});
