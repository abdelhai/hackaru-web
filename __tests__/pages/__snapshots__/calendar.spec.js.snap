// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar render correctly 1`] = `
<section>
  <date-header-stub
    date="Thu Jan 31 2019 01:23:45 GMT+0900 (GMT+09:00)"
    periodindex="1"
    periods="[object Object],[object Object]"
  />
   
  <div
    class="slider"
  >
    <section
      class="sticky"
    >
      <div
        class="headers-wrapper"
      >
        <div
          class="headers"
        >
          <calendar-day-header-stub
            day="Sun Jan 20 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Mon Jan 21 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Tue Jan 22 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Wed Jan 23 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Thu Jan 24 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Fri Jan 25 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Sat Jan 26 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
        </div>
        <div
          class="headers"
        >
          <calendar-day-header-stub
            day="Sun Jan 27 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Mon Jan 28 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Tue Jan 29 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Wed Jan 30 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Thu Jan 31 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Fri Feb 01 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Sat Feb 02 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
        </div>
        <div
          class="headers"
        >
          <calendar-day-header-stub
            day="Sun Feb 03 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Mon Feb 04 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Tue Feb 05 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Wed Feb 06 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Thu Feb 07 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Fri Feb 08 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
          <calendar-day-header-stub
            day="Sat Feb 09 2019 00:00:00 GMT+0900 (GMT+09:00)"
            locale="[object Object]"
          />
        </div>
      </div>
    </section>
     
    <section
      class="containers-wrapper"
    >
      <section
        class="containers"
      >
        <calendar-container-stub
          class="slider-item"
          days=""
        />
         
        <calendar-container-stub
          class="slider-item"
          days="Sun Jan 27 2019 00:00:00 GMT+0900 (GMT+09:00),Mon Jan 28 2019 00:00:00 GMT+0900 (GMT+09:00),Tue Jan 29 2019 00:00:00 GMT+0900 (GMT+09:00),Wed Jan 30 2019 00:00:00 GMT+0900 (GMT+09:00),Thu Jan 31 2019 00:00:00 GMT+0900 (GMT+09:00),Fri Feb 01 2019 00:00:00 GMT+0900 (GMT+09:00),Sat Feb 02 2019 00:00:00 GMT+0900 (GMT+09:00)"
        />
         
        <calendar-container-stub
          class="slider-item"
          days=""
        />
      </section>
    </section>
  </div>
</section>
`;
